### PR TITLE
[Docs] Added information to help users resolve IntelliJ issues.

### PR DESCRIPTION
**Pull request by devinbost**
I added some instructions to help developers resolve issues in IntelliJ that may occur when trying to run Pulsar unit tests. 
- [x] `doc`

### CODE REVIEW

**Review by Shoothzj — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -262,6 +262,15 @@ required plugins.
...
   * Sometimes it might help to mark a specific project ignored in IntelliJ Maven UI by right-clicking the project name and select **Ignore Projects** from the menu.
   * Currently, it is not always possible to run unit tests directly from the IDE because of the compilation issues. As a workaround, individual test classes can be run by using the `mvn test -Dtest=TestClassName` command.
+    
+* The above steps have all been performed, but a test still won't run.
+  * In this case, try the following steps:
+    * Close IntelliJ
+    * Run `mvn clean install -DskipTests` on the command line
+    * Reopen IntelliJ
+  * If that still doesn't work:
+    * Verify maven is using a supported version. Currently, the supported version of Maven is 3.6.1
```
Comment (nicoloboschi) [inline README.md:272]: we should be consistent with the maven enforcer plugin configuration

https://github.com/apache/pulsar/blob/bacac04f564536be9c30481d2cab88512906967e/pom.xml#L1650-L1652

**Review by nicoloboschi — CHANGES_REQUESTED** [review]

**Comment by Anonymitaet** [comment]
@devinbost when submitting a PR, can you provide doc-related info (tick the box) in the [PR description](https://github.com/apache/pulsar/blob/master/.github/PULL_REQUEST_TEMPLATE.md#documentation)? So that Bot can recognize and label your PR correspondingly. Or else Bot labels your PR with `doc-info-missing`.

Instructions of doc labels: https://docs.google.com/document/d/1Qw7LHQdXWBW9t2-r-A7QdFDBwmZh6ytB4guwMoXHqc0/edit#bookmark=id.5d66olk7l4oz 

image

File: README.md [hunk 2]
```diff
@@ -262,6 +262,15 @@ required plugins.
   * Use the "core-modules" profile if working on the Pulsar core modules since the source code for those modules can be compiled in IntelliJ.
   * Sometimes it might help to mark a specific project ignored in IntelliJ Maven UI by right-clicking the project name and select **Ignore Projects** from the menu.
   * Currently, it is not always possible to run unit tests directly from the IDE because of the compilation issues. As a workaround, individual test classes can be run by using the `mvn test -Dtest=TestClassName` command.
+    
+* The above steps have all been performed, but a test still won't run.
+  * In this case, try the following steps:
+    * Close IntelliJ
+    * Run `mvn clean install -DskipTests` on the command line
+    * Reopen IntelliJ
```
Comment (Anonymitaet) [inline README.md:270]:
```suggestion
    1. Close IntelliJ.
    2. Run `mvn clean install -DskipTests` on the command line.
    3. Reopen IntelliJ.
```

Use an ordered (numbered) list when the sequence is significant, for example, when writing procedures or ranking items. If the items in a list represent rules or other types of information that you want to refer to, you can refer to them by number. 

https://docs.google.com/document/d/1lc5j4RtuLIzlEYCBo97AC8-U_3Erzs_lxpkDuseU0n4/edit#bookmark=id.yoetnldi5amd

File: README.md (hunk 1 above)
Comment (Anonymitaet) [inline README.md:272]:
```suggestion
    1. Verify Maven is using a supported version. Currently, the supported version of Maven is 3.6.1.
```

Capitalize proper nouns correctly. 

https://docs.google.com/document/d/1lc5j4RtuLIzlEYCBo97AC8-U_3Erzs_lxpkDuseU0n4/edit#bookmark=id.je5thtkj42cb

File: README.md [hunk 3]
```diff
@@ -262,6 +262,15 @@ required plugins.
...
+  * If that still doesn't work:
+    * Verify maven is using a supported version. Currently, the supported version of Maven is 3.6.1
+    * Try "restart and clear caches" in IntelliJ and repeat the above steps to reload projects and generate sources.
```
Comment (Anonymitaet) [inline README.md:273]:
```suggestion
    2. Try "restart and clear caches" in IntelliJ and repeat the above steps to reload projects and generate sources.
```

File: README.md (hunk 1 above)
Comment (devinbost) [reply README.md:272]: Oh, we're requiring users to be running Maven 3.3.9?

**Review by nicoloboschi — APPROVED** [review]

**Review by Anonymitaet — APPROVED** [review]

**Pull request merged**